lib/web_catalog/api_extractor.es6.js

```diff
@@ -0,0 +1,406 @@
+      },
+    },
+    {
+      name: 'arrayMerge_',
+      description: 'Merge all given arrays into arr, no dupulicate allowed,' +
+        'boring and blacklisted properties are filtered',
+      code: function(arr) {
+        for (let i = 1; i < arguments.length; i++) {
+          for (let j = 0; j < arguments[i].length; j++) {
+            let str = this.cleanUp_(arguments[i][j]);
+            if (arr.indexOf(str) === -1 &&
+              ((this.buildInObjectProperties.indexOf(str) === -1 &&
+              this.buildInFunctionProperties.indexOf(str) === -1) ||
+              this.defaults.retainBuildInMembers) &&
+              !this.isPosotiveInt_(str)) {
+              arr.push(str);
+            }
+          }
+        }
+      },
+    },
+    {
+      name: 'setMinus_',
+      description: 'Return an array which is arr1 set minus arr2.',
+      code: function(arr1, arr2) {
+        let retArr = [];
+        for (let i = 0; i < arr1.length; i++) {
+          let str = this.cleanUp_(arr1[i]);
+          let cleanUpArr2 = arr2.map((str) => this.cleanUp_(str));
+          if (cleanUpArr2.indexOf(str) === -1) {
+            // Add str to restul Array if cannot find str in arr2.
+            retArr.push(str);
+          }
+        }
+        return retArr;
+      },
+    },
+    {
+      name: 'getFunctionBuildInProperties_',
+      description: 'Get list of properties from Function.prototype' +
+        'and push to buildInFunctionProperties.',
+      code: function getFunctionBlacklist(og) {
+        this.buildInFunctionProperties = ['prototype'];
+        this.metadata.functionId = og.lookup('Function', this.metadata.rootId);
+        this.metadata.functionProtoId = og.lookup('prototype',
+          this.metadata.functionId);
+        let functionProp = og.getObjectKeys(this.metadata.functionProtoId);
+        for (let i = 0; i < functionProp.length; i++) {
+          this.buildInFunctionProperties.push(this.cleanUp_(functionProp[i]));
+        }
+      },
+    },
+    {
+      name: 'getObjectBuildInProperties_',
+      description: 'Get list of properties from Object.prototype' +
+        'and push to buildInObjectProperties.',
+      code: function(og) {
+        this.buildInObjectProperties = [];
+        this.metadata.objectId = og.lookup('Object', this.metadata.rootId);
+        this.metadata.objectProtoId = og.lookup('prototype',
+          this.metadata.objectId);
+        let objectProp = og.getObjectKeys(this.metadata.objectProtoId);
+        for (let i = 0; i < objectProp.length; i++) {
+          this.buildInObjectProperties.push(this.cleanUp_(objectProp[i]));
+        }
+      },
+    },
+    {
+      name: 'getClassName_',
+      description: 'Search object id\'s prototype chain to find the the' +
+        'class it belongs to. The class is the prototype with a' +
+        'constructor property',
+      code: function(id, og) {
+        if (id === null || og.getType(id) !== 'object') return null;
```

Comment:
Yes, id should never be null. Then I think og.getType(id) !== 'object' should be enough, since only object can have constructor.